src/wallet/rpcwallet.cpp

```diff
@@ -2432,6 +2434,133 @@ UniValue getwalletinfo(const JSONRPCRequest& request)
     return obj;
 }
 
+UniValue listwallets(const JSONRPCRequest& request)
+{
+    if (request.fHelp || request.params.size() != 0)
+        throw std::runtime_error(
+            "listwallets\n"
+            "Returns a list of currently loaded wallets.\n"
+            "For full information on the wallet, use \"getwalletinfo\"\n"
+            "\nResult:\n"
+            "[\n"
+            "  \"walletname\": xxxxx,             (string) the wallet name\n"
+            "   ...\n"
+            "]\n"
+            "\nExamples:\n"
+            + HelpExampleCli("listwallets", "")
+            + HelpExampleRpc("listwallets", "")
+        );
+
+    LOCK2(cs_main, cs_wallets);
+
+    UniValue obj(UniValue::VARR);
+
+    for (CWalletRef pwallet : vpwallets) {
+
+        if (!EnsureWalletIsAvailable(pwallet, request.fHelp)) {
+            return NullUniValue;
+        }
+
+        LOCK(pwallet->cs_wallet);
+
+        obj.push_back(pwallet->GetName());
+    }
+
+    return obj;
+}
+
+UniValue loadwallet(const JSONRPCRequest& request)
```

Comment:
On further thought, I think I prefer the names `attachwallet` and `detachwallet`. That gives a bit more distinction from the dump/import RPCs and makes a stronger implication that the new `.dat` file is being loaded and attached as a separate wallet.